src/jit/gentree.cpp

```diff
@@ -8806,12 +8806,14 @@ unsigned GenTree::NumChildren()
         {
             return 3;
         }
-        // TODO-CQ: Check for possibilityof setting SetZero or similar intrinsics to unary operator type
-        // cannot set it globaly as one of GT_HWIntrinsics flags as this would break other instructionss
+#if FEATURE_HW_INTRINSICS
```

Comment:
I don't think I needed this, and I ended up implementing SetZero as well (for SSE).

Reply:
Wondering why you hit the assert and I didn't....

Reply:
I have used your implementation at the beginning but when testing optimized SetZero_ro in both Checked and Debug configs with AVX off via env variable and SSE2 legacy codegen on I have been hitting an assert directly below this code. When GT_HDIntrinsic was set to unary operator globally `(`HWIntrinsic      , GenTreeHWIntrinsic ,0,GTK_UNOP|GTK_BINOP|GTK_EXOP)` it messed up with all other instructions. I have not tested setting it during importation only for SetZero GenTree nodes.

Reply:
I don't think I tested with AVX off with the latest changes. I'll try that and see

Reply:
Ahh, forgot it AFAIR it was hit only for TYP_UBYTE arguments

Reply:
```
e:\src\ms\dotnet\coreclr>e:\src\ms\dotnet\coreclr\bin\tests\Windows_NT.x64.Debug\JIT\HardwareIntrinsics\X86\Sse2\SetZero_ro\SetZero_ro.cmd
BEGIN EXECUTION
 "e:\src\ms\dotnet\coreclr\bin\tests\Windows_NT.x64.Debug\Tests\Core_Root\corerun.exe" SetZero_ro.exe

Assert failure(PID 14660 [0x00003944], Thread: 14300 [0x37dc]): Assertion failed 'gtOp.gtOp1 != nullptr' in 'IntelHardwareIntrinsicTest.Program:Main(ref):int' (IL size 1116)

    File: e:\src\ms\dotnet\coreclr\src\jit\gentree.cpp Line: 8809
    Image: e:\src\ms\dotnet\coreclr\bin\tests\Windows_NT.x64.Debug\Tests\Core_Root\CoreRun.exe

Expected: 100
Actual: 123456789
END EXECUTION - FAILED
FAILED
```

Reply:
I see, it happens when doing loop hoisting.